rtengine/CA_correct_RT.cc

```diff
@@ -1153,14 +1184,27 @@ void RawImageSource::CA_correct_RT(const bool autoCA, const double cared, const
         }
 
         // clean up
-        free(buffer);
+        free(bufferThr);
+    }
+
+    if(autoCA && fitParamsTransfer && fitParamsOut) {
+        // store calculated parameters
+        int index = 0;
+        for(int c = 0; c < 2; ++c) {
+            for(int d = 0; d < 2; ++d) {
+                for(int e = 0; e < 16; ++e) {
+                    fitParamsTransfer[index++] = fitparams[c][d][e];
+                }
+            }
+        }
     }
 
-    free(Gtmp);
-    free(blockwt);
-    free(RawDataTmp);
+    if(freeBuffer) {
+        free(buffer);
+    }
 
     if(plistener) {
         plistener->setProgress(1.0);
     }
+    return buffer;
```

Comment:
Better return `nullptr` if you freed the pointer.

Reply:
Done